### DISCUSSION TITLE
For accessing GitHub from the command line, which is the most recommended way of authentication

### DISCUSSION THREAD

**Discussion by songyuc** [community/community#67749] [New to GitHub]
Hello GitHub community,

I'm relatively new to Git and GitHub, and I'm currently working on various Python projects. I aim to use Git for version control across all these projects.

I've noticed that there are multiple methods for authenticating GitHub from the command line, such as GitHub CLI, [Git Credential Manager](https://github.com/GitCredentialManager/git-credential-manager/blob/main/README.md), HTTPS and SSH. Given the constant updates and changes in best practices, I'm a bit confused about which method is most recommended for secure and efficient workflow up to date.

Could anyone shed some light on this?

Looking forward to your insights!

Best regards,
Yucheng Song

**Comment by Abhiram-J** [comment]
Hi songyuc,
Which Operating System are you using ?

**Reply by songyuc** [reply]
Hi, Abhiram,
I am working on Windows and Linux together.

**Reply by Abhiram-J** [reply]
Hi songyuc,
Same here, I too use both of them, one for coding and other for entertainment purposes.
As I use Linux most of the time, I prefer the SSH method its more comfortable for basic git commands and pushing to remote repositories. 
I use the github GUI in browser for other things like PRs and Forks. 
You can also use PRs from terminal using github CLI.

**Reply by songyuc** [reply]
I really appreciate your suggestions. I will try out the methods you've recommended. Thank you very much.
